File: leetCode/top150/Arrays_Strings/1_TwoSum.py
from collections import defaultdict

class Solution(object):
    def twoSum(self, nums, target):
        prevMap = defaultdict(int) # num : index

        for i in range(len(nums)):
            n = nums[i]
            diff = target - n
            if diff in prevMap:
                return [prevMap[diff], i]
            prevMap[n] = i
        

File: AlgorithmPractice/150/HashMaps/1_TwoSum.ts
function twoSum(nums: number[], target: number): number[] {
    const prevMap = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const diff = target - nums[i];

        if (diff in prevMap) {
            return [prevMap[diff], i];
        }

        prevMap[nums[i]] = i;
    }
};